app/models/order.rb

```diff
@@ -0,0 +1,23 @@
+class Order < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :event
+
+  has_many :order_items, inverse_of: :order, dependent: :destroy
+
+  validates :user, :event, presence: true
+
+  delegate :name, to: :event, prefix: true
+
+  accepts_nested_attributes_for :order_items
+
+  def self.build_from_event(event)
+    order_items = event.ticket_types.map do |ticket_type|
```

Comment:
This can be a bit shorter. Take advantage of Rails associations magic. Since an Order has many OrderItem, you have access to the methods `order.order_items.build` and `order.order_items.create`.